yarn-project/aztec-node/src/aztec-node/server.ts

```diff
@@ -172,6 +172,7 @@ export class AztecNodeService implements AztecNode, AztecNodeAdmin, Traceable {
       prefilledPublicData?: PublicDataTreeLeaf[];
     } = {},
   ): Promise<AztecNodeService> {
+    const config = { ...inputConfig }; // Copy the config so we dont mutate the input object
```

Comment:
IIRC nested objects (l1 contract addresses) are not properly cloned this way, they can still be mutated if the original object is